File: Bridge/cpp-source/wheels/PlaneWheel.cpp
//
// Created by Pavel Akhtyamov on 2019-03-26.
//

#include <PlaneWheel.h>

void PlaneWheel::PushWheel() {
  axle_->Brake(0.1);
  axle_->MoveDown(0.1);
}

void PlaneWheel::PullWheel() {
  axle_->Accelerate(0.1);
  axle_->MoveUp(0.1);
}

void PlaneWheel::PressGasPedal() {
  axle_->Accelerate(0.1);
}

void PlaneWheel::PressBreakPedal() {
  axle_->Brake(0.1);
  axle_->MoveDown(0.1);
}

void PlaneWheel::PressLeftButton() {
  axle_->Beep();
}

void PlaneWheel::PressRightButton() {
  axle_->PrintInfo();
}

void PlaneWheel::MoveRightWheel(float angle) {
  axle_->MoveRight(angle);
}

void PlaneWheel::MoveLeftWheel(float angle) {
  axle_->MoveLeft(angle);
}

void PlaneWheel::PressUpButton() {}

void PlaneWheel::PressDownButton() {}

void PlaneWheel::PrintInfo() {
  axle_->PrintInfo();
}

PlaneWheel::PlaneWheel(std::unique_ptr<Axle> axle) : SteeringWheel(std::move(axle)) {

}

File: Bridge/python-source/wheels/PlaneWheel.py
from __future__ import annotations

from .SteeringWheel import SteeringWheel


class PlaneWheel(SteeringWheel):
    """Штурвал: движение самого штурвала задаёт высоту."""

    STEP = 0.1

    def push_wheel(self) -> None:
        self._axle.move_down(self.STEP)

    def pull_wheel(self) -> None:
        self._axle.move_up(self.STEP)

    def press_gas_pedal(self) -> None:
        self._axle.accelerate(self.STEP)

    def press_brake_pedal(self) -> None:
        self._axle.brake(self.STEP)

    def press_left_button(self) -> None:
        self._axle.move_left(self.STEP)

    def press_right_button(self) -> None:
        self._axle.move_right(self.STEP)

    def press_up_button(self) -> None:
        self._axle.beep()

    def press_down_button(self) -> None:
        self._axle.beep()
